storage/leveldb/leveldb.go

```diff
@@ -57,32 +59,34 @@ func NewDB(path string, batchDelaySeconds int, maxBatchSize int, maxOpenFiles in
 	}
 
 	bldb := &baseLevelDb{
-		db: db,
+		db:   db,
+		path: path,
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
 	dbStore := &DB{
 		baseLevelDb:       bldb,
-		path:              path,
 		maxBatchSize:      maxBatchSize,
 		batchDelaySeconds: batchDelaySeconds,
 		sizeBatch:         0,
-		dbClosed:          make(chan struct{}),
+		cancel:            cancel,
 	}
 
 	dbStore.batch = dbStore.createBatch()
 
-	go dbStore.batchTimeoutHandle()
+	go dbStore.batchTimeoutHandle(ctx)
 
 	runtime.SetFinalizer(dbStore, func(db *DB) {
 		_ = db.Close()
 	})
 
-	log.Debug("opened level db persister", "path", path)
+	crtCounter := atomic.AddUint32(&loggingDBCounter, 1)
+	log.Debug("opened level db persister", "path", path, "created pointer", fmt.Sprintf("%p", bldb.db), "global db counter", crtCounter)
```

Comment:
could have used `logDebug`

Reply:
right, removed logDebug pointer in order to keep only one log pointer on the package